cmd/utils/flags.go

```diff
@@ -941,10 +943,11 @@ func SetEthConfig(ctx *cli.Context, stack *node.Node, cfg *eth.Config) {
 	case ctx.GlobalBool(LightModeFlag.Name):
 		cfg.SyncMode = downloader.LightSync
 	}
-	if ctx.GlobalIsSet(LightServFlag.Name) {
+	
+	if ctx.GlobalIsSet(LightServFlag.Name) && cfg.SyncMode != downloader.LightSync {
 		cfg.LightServ = ctx.GlobalInt(LightServFlag.Name)
 	}
-	if ctx.GlobalIsSet(LightPeersFlag.Name) {
+	if ctx.GlobalIsSet(LightPeersFlag.Name) && cfg.SyncMode != downloader.LightSync {
```

Comment:
So, there aren't really the best because it introduces undefined behavior when combo-flags are set. We need to forbid these combos altogether above during exclusivity checks.